scripts/regression_test_python.py

```diff
@@ -72,6 +72,8 @@ def __init__(self, scale_factor):
 
     def get_tables(self, convertor) -> Dict[str, Any]:
         res = {}
+        # Without this, the DECIMAL type analyzed could be too small depending on the distribution
+        self.conn.execute("set global pandas_analyze_sample=2048")
```

Comment:
This is caused by the pandas analyzer not seeing decimals with a big enough width.
One other way we can fix this is by making the detected type by the analyzer wider

So whenever we find a DECIMAL with a width that would use a `int16_t` internally, we would set the detected width to the max of that physical type (4 in this case)

Reply:
Seems that this bandaid doesn't work even, I'll try the widening, that sounds like a better solution in general